smart-contracts/wasm-transform/src/validate.rs

```diff
@@ -934,7 +946,7 @@ pub fn validate_module(
     // The global section is valid as long as it's well-formed.
     // We already check that all the globals are initialized with
     // correct expressions.
-    let global: GlobalSection = parse_sec_with_default(EMPTY_CTX, &skeleton.global)?;
+    let global: GlobalSection = parse_sec_with_default(config, &skeleton.global)?;
     ensure!(
         global.globals.len() <= MAX_NUM_GLOBALS,
         "The number of globals must not exceed {}.",
```

Comment:
Don't we also need the non-empty context when parsing the functions section further below?
I expect that we do, since (to my understanding) it can contain instructions

Reply:
Okay, having read a bit more about how the binary representation is, I take this back.
I see now that all the instructions are put in the code section

Reply:
Function section does not contain the code. It contains the type declarations of functions, i.e., 

```
void myFunction();
```

The code section contains instructions, and we pass in the relevant context to the instruction parser

Reply:
I submitted my comment without seeing your own second comment. All good then.